barretenberg/cpp/src/barretenberg/vm/tests/avm_arithmetic.test.cpp

```diff
@@ -441,6 +497,41 @@ TEST_F(AvmArithmeticTestsFF, mixedOperationsWithError)
     validate_trace_proof(std::move(trace));
 }
 
+// Test of equality on FF elements
+TEST_F(AvmArithmeticTestsFF, equality)
+{
+    // Pick a field-sized number
+    FF elem = FF::modulus - FF::one();
+    trace_builder.calldata_copy(0, 3, 0, std::vector<FF>{ elem, elem, 1 });
+    trace_builder.op_eq(0, 1, 2, AvmMemoryTag::FF); // Memory Layout [q - 1, q -1, 1,0..]
+    trace_builder.return_op(0, 3);
+    auto trace = trace_builder.finalize();
+
+    auto alu_row_index = common_validate_eq(trace, elem, elem, FF(1), FF(0), FF(1), FF(2), AvmMemoryTag::FF);
+    auto alu_row = trace.at(alu_row_index);
+
+    EXPECT_EQ(alu_row.avm_alu_alu_ff_tag, FF(1));
```

Comment:
Isn-t this check performed as part of the common_validate_eq (common arithmetic function)?